calledudeBot/Services/SongRequester.cs

```diff
@@ -0,0 +1,39 @@
+﻿using calledudeBot.Chat;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace calledudeBot.Services
+{
+    public sealed class SongRequester
+    {
+        private const string _songRequestLink = "https://osu.ppy.sh/api/get_beatmaps?k={0}&b={1}";
+        private readonly string _osuAPIToken;
+
+        public SongRequester(string osuAPIToken)
+        {
+            _osuAPIToken = osuAPIToken;
+        }
+
+        //[http://osu.ppy.sh/b/795232 fhana - Wonder Stella [Stella]]
+        public async Task RequestSong(Message message)
+        {
+            var idx = message.Content.IndexOf("/b/") + "/b/".Length;
+            var num = message.Content.Skip(idx).TakeWhile(c => char.IsNumber(c));
```

Comment:
Would `char.IsDigit()` fit better than `char.IsNumber()`? I assume things like `½` aren't supposed to be valid.

Reply:
TIL